src/main/scala/mesosphere/marathon/core/matcher/manager/impl/OfferMatcherManagerActor.scala

```diff
@@ -132,14 +134,14 @@ private class OfferMatcherManagerActor private (
       context.system.scheduler.scheduleOnce(
         clock.now().until(deadline),
         self,
-        OfferMatcher.MatchedTasks(offer.getId, Seq.empty))
+        MatchTimeout(offer.getId))
 
       // process offer for the first time
       scheduleNextMatcherOrFinish(data)
   }
 
   private[this] def receiveMatchedTasks: Receive = {
-    case OfferMatcher.MatchedTasks(offerId, addedTasks) =>
+    case OfferMatcher.MatchedTasks(offerId, addedTasks, _) =>
```

Comment:
We should evaluate the `resendThisOffer` flag here. If any sub matcher responded with `resendThisOffer=true`, the manager should do that as well.